samples/tacos/src/main/kotlin/com/slack/circuit/tacos/OrderTacosCircuit.kt

```diff
@@ -0,0 +1,356 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.slack.circuit.runtime.CircuitUiEvent
+import com.slack.circuit.runtime.CircuitUiState
+import com.slack.circuit.runtime.Screen
+import com.slack.circuit.runtime.presenter.Presenter
+import com.slack.circuit.tacos.model.Diet
+import com.slack.circuit.tacos.model.Ingredient
+import com.slack.circuit.tacos.step.ConfirmationOrderStep
+import com.slack.circuit.tacos.step.ConfirmationUi
+import com.slack.circuit.tacos.step.FillingsOrderStep
+import com.slack.circuit.tacos.step.FillingsUi
+import com.slack.circuit.tacos.step.OrderStep
+import com.slack.circuit.tacos.step.SummaryOrderStep
+import com.slack.circuit.tacos.step.SummaryUi
+import com.slack.circuit.tacos.step.ToppingsOrderStep
+import com.slack.circuit.tacos.step.ToppingsUi
+import com.slack.circuit.tacos.ui.theme.TacoTheme
+import java.math.BigDecimal
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.persistentSetOf
+import kotlinx.collections.immutable.toImmutableSet
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+object OrderTacosScreen : Screen {
+  data class State(
+    @StringRes val headerResId: Int,
+    val orderCost: BigDecimal,
+    val stepState: OrderStep.State,
+    val isPreviousVisible: Boolean,
+    val isNextEnabled: Boolean,
+    val isNextVisible: Boolean,
+    val isFooterVisible: Boolean,
+    val eventSink: (Event) -> Unit
+  ) : CircuitUiState
+
+  sealed interface Event : CircuitUiEvent {
+    val indexModifier: Int
+      get() = 0
+
+    object Previous : Event {
+      override val indexModifier: Int = -1
+    }
+    object Next : Event {
+      override val indexModifier: Int = 1
+    }
+    object ProcessOrder : Event {
+      override val indexModifier: Int = 1
+    }
+  }
+}
+
+@Immutable
+data class OrderDetails(
+  val filling: Ingredient = Ingredient(""),
+  val toppings: Set<Ingredient> = persistentSetOf(),
+  val baseCost: BigDecimal = BigDecimal("9.99"), // Default taco price
+  val ingredientsCost: BigDecimal = BigDecimal.ZERO
+)
+
+/** List of wizard steps (in order!) */
+private val orderSteps =
+  persistentListOf(FillingsOrderStep, ToppingsOrderStep, ConfirmationOrderStep, SummaryOrderStep)
+
+internal class OrderTacosPresenter(
+  private val fillingsProducer: OrderStep.StateProducer<FillingsOrderStep.State>,
+  private val toppingsProducer: OrderStep.StateProducer<ToppingsOrderStep.State>,
+  private val confirmationProducer: OrderStep.StateProducer<ConfirmationOrderStep.OrderState>,
+  private val summaryProducer: OrderStep.StateProducer<SummaryOrderStep.SummaryState>,
+) : Presenter<OrderTacosScreen.State> {
+  @Composable override fun present(): OrderTacosScreen.State = presentInternal()
+
+  @Composable
+  internal fun presentInternal(
+    initialStep: OrderStep = FillingsOrderStep,
+    initialOrderDetails: OrderDetails = OrderDetails(),
+  ): OrderTacosScreen.State {
```

Comment:
it feels like an anti-pattern for circuit that we're not testing from the `present()` function alone 🤔. Could we do this another way? Maybe make an overloaded constructor that takes these initial args. I think we actually _have_ to do that, because this function can be re-called multiple times so `initial` isn't the right way to think of it here

Reply:
Sure, I can look at overloading the constructor and see where that gets me